packages/daemon/src/daemon.js

```diff
@@ -39,33 +44,149 @@ const sinkError = error => {
 
 /**
  * @param {import('../index.js').Locator} locator
+ * @param {string} sha512
  */
-const makeWorker = async locator => {
-  // @ts-ignore Node.js crypto does in fact have randomUUID.
-  const uuid = await crypto.randomUUID();
-  const workerCachePath = path.join(locator.cachePath, uuid);
-  await fs.promises.mkdir(workerCachePath, { recursive: true });
-  const logPath = path.join(workerCachePath, 'worker.log');
-  const output = fs.openSync(logPath, 'w');
-  const child = popen.fork(endoWorkerPath, [uuid, workerCachePath], {
-    stdio: ['ignore', output, output, 'pipe', 'ipc'],
+const provideReadableSha512 = (locator, sha512) => {
```

Comment:
as we discussed, seems to make a new one always. should be called `makeReadableSha512`